src/rosdiscover/observer/observer.py

```diff
@@ -23,7 +23,23 @@ def for_container(cls,
                       container: str,
                       config: 'Config',
                       ) -> Iterator['Observer']:
-        """Constructs and interpreter for a given running container"""
+        """Constructs and interpreter for a given running container
```

Comment:
A couple of minor issues: "and" should probably be "an", and "interpreter" should probably be "observer".

Reply:
More importantly, there isn't any need for this to be a `contextmanager`. This method can just return an `Observer` rather than an `Iterator['Observer']` since there is no associated resource management.

Reply:
Done